src/nnet3bin/nnet3-discriminative-subset-egs.cc

```diff
@@ -0,0 +1,102 @@
+// nnet3bin/nnet3-discriminative-subset-egs.cc
+
+// Copyright 2012-2015  Johns Hopkins University (author:  Daniel Povey)
+//                2014  Vimal Manohar
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "nnet3/nnet-discriminative-example.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace kaldi::nnet3;
+    typedef kaldi::int32 int32;
+    typedef kaldi::int64 int64;
+
+    const char *usage =
+        "Creates a random subset of the input examples, of a specified size.\n"
+        "Uses no more memory than the size of the subset.\n"
+        "\n"
+        "Usage:  nnet3-discriminative-subset-egs [options] <degs-rspecifier> [<degs-wspecifier2> ...]\n"
+        "\n"
+        "e.g.\n"
+        "nnet3-discriminative-subset-egs [args] ark:- | nnet-discriminative-subset-egs --n=1000 ark:- ark:subset.degs\n";
```

Comment:
there seems to be something wrong in this e.g. line.  it's too long, it has nnet-discr.... as a name, and the pipe does not really make sense.